libsolidity/ast/Types.cpp

```diff
@@ -735,19 +736,43 @@ string FixedPointType::toString(bool) const
 	return prefix + util::toString(m_totalBits) + "x" + util::toString(m_fractionalDigits);
 }
 
+optional<FixedPointType> FixedPointType::fromString(std::string const& _type)
+{
+	smatch matches;
+	if (regex_match(_type, matches, regex{"(.?)fixed(\\d+)x(\\d+)"}))
+	{
+		Modifier modifier{Modifier::Signed};
+		if (matches[1].str() == "u")
+			modifier = Modifier::Unsigned;
+		unsigned totalBits = static_cast<unsigned>(std::stoi(matches[2].str()));
+		unsigned fractionalDigits = static_cast<unsigned>(std::stoi(matches[3].str()));
+		return FixedPointType(totalBits, fractionalDigits, modifier);
```

Comment:
I would like to reuse the `FixedPointType` class here. But the original design of AST types prevent its construction here. Would it be ok to allow them to be copy constructable?

Reply:
Without having seen any context: Types are only used as pointers. Use the TypeProvider to create them.

Reply:
Ok, I changed the code to use the `TypeProvider`.